tests/subscriptions/test_scheduler_processing_strategy.py

```diff
@@ -200,3 +204,124 @@ def test_tick_buffer_wait_slowest() -> None:
     assert next_step.submit.call_count == 1
     assert next_step.submit.call_args_list == [mock.call(messages[0])]
     assert metrics_backend.calls == []
+
+
+def make_message_for_next_step(
+    message: Message[Tick], should_commit: bool
+) -> Message[CommittableTick]:
+    return Message(
+        message.partition,
+        message.offset,
+        CommittableTick(message.payload, should_commit),
+        message.timestamp,
+        message.next_offset,
+    )
+
+
+def test_provide_commit_strategy() -> None:
+    epoch = datetime(1970, 1, 1)
+    next_step = mock.Mock()
+    strategy = ProvideCommitStrategy(2, next_step)
+
+    topic = Topic("messages")
+    partition = Partition(topic, 0)
+
+    # First message for partition 0 -> do not commit offset
+    message_p0 = Message(
+        partition,
+        1,
+        Tick(
+            0,
+            offsets=Interval(1, 2),
+            timestamps=Interval(
+                epoch + timedelta(seconds=1), epoch + timedelta(seconds=5)
+            ),
+        ),
+        epoch,
+        2,
+    )
+
+    strategy.submit(message_p0)
+    assert next_step.submit.call_args_list == [
+        mock.call(make_message_for_next_step(message_p0, False))
+    ]
+
+    next_step.reset_mock()
+
+    # Partition 1, same timestamp -> now we commit the offset
+    message_p1 = Message(
+        partition,
+        2,
+        Tick(
+            1,
+            offsets=Interval(11, 12),
+            timestamps=Interval(
+                epoch + timedelta(seconds=1), epoch + timedelta(seconds=5)
+            ),
+        ),
+        epoch,
+        3,
+    )
+
+    strategy.submit(message_p1)
+
+    assert next_step.submit.call_args_list == [
+        mock.call(make_message_for_next_step(message_p1, True))
```

Comment:
I'd beef up this test as it test one simple use case.
Two messages for two partitions with timestamps that are perfectly aligned.
I think there are quite a few interesting cases to test.
- multiple messages for one partitions before you see the first message of the second
- time stamps not aligned. Like partition1: 1-5,6-9,10-11 and partition2: 1-2,3-5,6-12
- Multiple updates of the watermark causing more than one commit only. Like starting with the case you are already testing, then keep submitting till you see the second commit to ensure watermarks are updated properly.